pennylane_lightning_gpu/src/simulator/StateVectorCudaMPI.hpp

```diff
@@ -0,0 +1,1461 @@
+// Copyright 2022 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @file StateVectorCudaMPI.hpp
+ */
+#pragma once
+
+#include <functional>
+#include <numeric>
+#include <random>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include <cuComplex.h> // cuDoubleComplex
+#include <cuda.h>
+#include <custatevec.h> // custatevecApplyMatrix
```

Comment:
Nice, please add as much as possible, the name of the functions used beside the header inclusion.
This improves future development and maintainability.

Reply:
Sure!